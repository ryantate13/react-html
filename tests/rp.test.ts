import {rp} from '../src';

describe('rp', () => {
    it('renders with children', () => {
        expect(rp({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(rp({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(rp()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(rp({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(rp({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(rp({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(rp({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(rp({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(rp({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(rp({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(rp({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(rp({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(rp({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(rp({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(rp({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(rp({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(rp({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(rp({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(rp({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(rp({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(rp({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(rp({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(rp({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(rp({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(rp({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(rp({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(rp({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(rp({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(rp({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(rp({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(rp({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(rp({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(rp({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(rp({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(rp({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(rp({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(rp({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(rp({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(rp({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(rp({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(rp({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(rp({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(rp({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(rp({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(rp({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(rp({translate: "no"})).toMatchSnapshot();
    });
});