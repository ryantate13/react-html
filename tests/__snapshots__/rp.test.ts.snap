// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rp renders arrays of children 1`] = `
<rp>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</rp>
`;

exports[`rp renders with accessKey="a,b,c,d" 1`] = `
<rp
  accessKey="a,b,c,d"
/>
`;

exports[`rp renders with autoCapitalize="characters" 1`] = `
<rp
  autoCapitalize="characters"
/>
`;

exports[`rp renders with autoCapitalize="none" 1`] = `
<rp
  autoCapitalize="none"
/>
`;

exports[`rp renders with autoCapitalize="off" 1`] = `
<rp
  autoCapitalize="off"
/>
`;

exports[`rp renders with autoCapitalize="on" 1`] = `
<rp
  autoCapitalize="on"
/>
`;

exports[`rp renders with autoCapitalize="sentences" 1`] = `
<rp
  autoCapitalize="sentences"
/>
`;

exports[`rp renders with autoCapitalize="words" 1`] = `
<rp
  autoCapitalize="words"
/>
`;

exports[`rp renders with children 1`] = `
<rp>
  TEST
</rp>
`;

exports[`rp renders with className="foo bar baz" 1`] = `
<rp
  className="foo bar baz"
/>
`;

exports[`rp renders with contentEditable="false" 1`] = `
<rp
  contentEditable="false"
/>
`;

exports[`rp renders with contentEditable="true" 1`] = `
<rp
  contentEditable="true"
/>
`;

exports[`rp renders with dir="auto" 1`] = `
<rp
  dir="auto"
/>
`;

exports[`rp renders with dir="ltr" 1`] = `
<rp
  dir="ltr"
/>
`;

exports[`rp renders with dir="rtl" 1`] = `
<rp
  dir="rtl"
/>
`;

exports[`rp renders with draggable="false" 1`] = `
<rp
  draggable="false"
/>
`;

exports[`rp renders with draggable="true" 1`] = `
<rp
  draggable="true"
/>
`;

exports[`rp renders with empty attributes 1`] = `<rp />`;

exports[`rp renders with hidden=false 1`] = `
<rp
  hidden={false}
/>
`;

exports[`rp renders with hidden=true 1`] = `
<rp
  hidden={true}
/>
`;

exports[`rp renders with id="TEST" 1`] = `
<rp
  id="TEST"
/>
`;

exports[`rp renders with inputMode="decimal" 1`] = `
<rp
  inputMode="decimal"
/>
`;

exports[`rp renders with inputMode="email" 1`] = `
<rp
  inputMode="email"
/>
`;

exports[`rp renders with inputMode="none" 1`] = `
<rp
  inputMode="none"
/>
`;

exports[`rp renders with inputMode="numeric" 1`] = `
<rp
  inputMode="numeric"
/>
`;

exports[`rp renders with inputMode="search" 1`] = `
<rp
  inputMode="search"
/>
`;

exports[`rp renders with inputMode="tel" 1`] = `
<rp
  inputMode="tel"
/>
`;

exports[`rp renders with inputMode="text" 1`] = `
<rp
  inputMode="text"
/>
`;

exports[`rp renders with inputMode="url" 1`] = `
<rp
  inputMode="url"
/>
`;

exports[`rp renders with is="foo" 1`] = `
<rp
  is="foo"
/>
`;

exports[`rp renders with itemID="https://google.com/" 1`] = `
<rp
  itemID="https://google.com/"
/>
`;

exports[`rp renders with itemProp="https://google.com/" 1`] = `
<rp
  itemProp="https://google.com/"
/>
`;

exports[`rp renders with itemRef="foo bar" 1`] = `
<rp
  itemRef="foo bar"
/>
`;

exports[`rp renders with itemScope=false 1`] = `
<rp
  itemScope={false}
/>
`;

exports[`rp renders with itemScope=true 1`] = `
<rp
  itemScope={true}
/>
`;

exports[`rp renders with itemType="https://google.com/" 1`] = `
<rp
  itemType="https://google.com/"
/>
`;

exports[`rp renders with lang="" 1`] = `
<rp
  lang=""
/>
`;

exports[`rp renders with lang="es-419" 1`] = `
<rp
  lang="es-419"
/>
`;

exports[`rp renders with slot="TEST" 1`] = `
<rp
  slot="TEST"
/>
`;

exports[`rp renders with spellCheck="false" 1`] = `
<rp
  spellCheck="false"
/>
`;

exports[`rp renders with spellCheck="true" 1`] = `
<rp
  spellCheck="true"
/>
`;

exports[`rp renders with style={"color":"#000000"} 1`] = `
<rp
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`rp renders with tabIndex=1 1`] = `
<rp
  tabIndex={1}
/>
`;

exports[`rp renders with title="TEST" 1`] = `
<rp
  title="TEST"
/>
`;

exports[`rp renders with translate="no" 1`] = `
<rp
  translate="no"
/>
`;

exports[`rp renders with translate="yes" 1`] = `
<rp
  translate="yes"
/>
`;

exports[`rp renders without attributes 1`] = `<rp />`;
